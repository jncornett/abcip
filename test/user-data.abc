c ( a.file="tmp/data.txt" )
c ( b.data="The quick brown fragments jumped over the lazy packets. " )

a(syn)
b(syn, ack)
a(ack)

a(ack, len="max")

b(ack, len=35)
b(ack, len=25)
b(ack, len=15)

a(fin,ack)
b(fin,ack)
a(ack)

