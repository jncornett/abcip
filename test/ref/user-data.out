IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [S], cksum 0xb3d9 (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0001 0000 0000 5002  ..............P.
	0x0030:  2000 b3d9 0000                           ......
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [S.], cksum 0xb3c7 (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0001 0000 0002 5012  ..............P.
	0x0030:  2000 b3c7 0000                           ......
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xb3c8 (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 b3c8 0000                           ......
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 975)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x301a (correct), seq 1:936, ack 1, win 8192, length 935
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  03cf 0003 0000 4006 5913 0a01 0203 0a09  ......@.Y.......
	0x0020:  0807 bdec 0008 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 301a 0000 2754 7761 7320 6272 696c  ..0...'Twas.bril
	0x0040:  6c69 672c 2061 6e64 2074 6865 2073 6c69  lig,.and.the.sli
	0x0050:  7468 7920 746f 7665 730a 4469 6420 6779  thy.toves.Did.gy
	0x0060:  7265 2061 6e64 2067 696d 626c 6520 696e  re.and.gimble.in
	0x0070:  2074 6865 2077 6162 653b 0a41 6c6c 206d  .the.wabe;.All.m
	0x0080:  696d 7379 2077 6572 6520 7468 6520 626f  imsy.were.the.bo
	0x0090:  726f 676f 7665 732c 0a41 6e64 2074 6865  rogoves,.And.the
	0x00a0:  206d 6f6d 6520 7261 7468 7320 6f75 7467  .mome.raths.outg
	0x00b0:  7261 6265 2e0a 0a22 4265 7761 7265 2074  rabe..."Beware.t
	0x00c0:  6865 204a 6162 6265 7277 6f63 6b2c 206d  he.Jabberwock,.m
	0x00d0:  7920 736f 6e21 0a54 6865 206a 6177 7320  y.son!.The.jaws.
	0x00e0:  7468 6174 2062 6974 652c 2074 6865 2063  that.bite,.the.c
	0x00f0:  6c61 7773 2074 6861 7420 6361 7463 6821  laws.that.catch!
	0x0100:  0a42 6577 6172 6520 7468 6520 4a75 626a  .Beware.the.Jubj
	0x0110:  7562 2062 6972 642c 2061 6e64 2073 6875  ub.bird,.and.shu
	0x0120:  6e0a 5468 6520 6672 756d 696f 7573 2042  n.The.frumious.B
	0x0130:  616e 6465 7273 6e61 7463 6821 220a 0a48  andersnatch!"..H
	0x0140:  6520 746f 6f6b 2068 6973 2076 6f72 7061  e.took.his.vorpa
	0x0150:  6c20 7377 6f72 6420 696e 2068 616e 643a  l.sword.in.hand:
	0x0160:  0a4c 6f6e 6720 7469 6d65 2074 6865 206d  .Long.time.the.m
	0x0170:  616e 786f 6d65 2066 6f65 2068 6520 736f  anxome.foe.he.so
	0x0180:  7567 6874 2d2d 0a53 6f20 7265 7374 6564  ught--.So.rested
	0x0190:  2068 6520 6279 2074 6865 2054 756d 7475  .he.by.the.Tumtu
	0x01a0:  6d20 7472 6565 2c0a 416e 6420 7374 6f6f  m.tree,.And.stoo
	0x01b0:  6420 6177 6869 6c65 2069 6e20 7468 6f75  d.awhile.in.thou
	0x01c0:  6768 742e 0a0a 416e 6420 6173 2069 6e20  ght...And.as.in.
	0x01d0:  7566 6669 7368 2074 686f 7567 6874 2068  uffish.thought.h
	0x01e0:  6520 7374 6f6f 642c 0a54 6865 204a 6162  e.stood,.The.Jab
	0x01f0:  6265 7277 6f63 6b2c 2077 6974 6820 6579  berwock,.with.ey
	0x0200:  6573 206f 6620 666c 616d 652c 0a43 616d  es.of.flame,.Cam
	0x0210:  6520 7768 6966 666c 696e 6720 7468 726f  e.whiffling.thro
	0x0220:  7567 6820 7468 6520 7475 6c67 6579 2077  ugh.the.tulgey.w
	0x0230:  6f6f 642c 0a41 6e64 2062 7572 626c 6564  ood,.And.burbled
	0x0240:  2061 7320 6974 2063 616d 6521 0a0a 4f6e  .as.it.came!..On
	0x0250:  652c 2074 776f 2120 4f6e 652c 2074 776f  e,.two!.One,.two
	0x0260:  2120 616e 6420 7468 726f 7567 6820 616e  !.and.through.an
	0x0270:  6420 7468 726f 7567 680a 5468 6520 766f  d.through.The.vo
	0x0280:  7270 616c 2062 6c61 6465 2077 656e 7420  rpal.blade.went.
	0x0290:  736e 6963 6b65 722d 736e 6163 6b21 0a48  snicker-snack!.H
	0x02a0:  6520 6c65 6674 2069 7420 6465 6164 2c20  e.left.it.dead,.
	0x02b0:  616e 6420 7769 7468 2069 7473 2068 6561  and.with.its.hea
	0x02c0:  640a 4865 2077 656e 7420 6761 6c75 6d70  d.He.went.galump
	0x02d0:  6869 6e67 2062 6163 6b2e 0a0a 2241 6e64  hing.back..."And
	0x02e0:  2068 6173 7420 7468 6f75 2073 6c61 696e  .hast.thou.slain
	0x02f0:  2074 6865 204a 6162 6265 7277 6f63 6b3f  .the.Jabberwock?
	0x0300:  0a43 6f6d 6520 746f 206d 7920 6172 6d73  .Come.to.my.arms
	0x0310:  2c20 6d79 2062 6561 6d69 7368 2062 6f79  ,.my.beamish.boy
	0x0320:  210a 4f20 6672 6162 6a6f 7573 2064 6179  !.O.frabjous.day
	0x0330:  2120 4361 6c6c 6f6f 6821 2043 616c 6c61  !.Callooh!.Calla
	0x0340:  7921 220a 4865 2063 686f 7274 6c65 6420  y!".He.chortled.
	0x0350:  696e 2068 6973 206a 6f79 2e0a 0a27 5477  in.his.joy...'Tw
	0x0360:  6173 2062 7269 6c6c 6967 2c20 616e 6420  as.brillig,.and.
	0x0370:  7468 6520 736c 6974 6879 2074 6f76 6573  the.slithy.toves
	0x0380:  0a44 6964 2067 7972 6520 616e 6420 6769  .Did.gyre.and.gi
	0x0390:  6d62 6c65 2069 6e20 7468 6520 7761 6265  mble.in.the.wabe
	0x03a0:  3b0a 416c 6c20 6d69 6d73 7920 7765 7265  ;.All.mimsy.were
	0x03b0:  2074 6865 2062 6f72 6f67 6f76 6573 2c0a  .the.borogoves,.
	0x03c0:  416e 6420 7468 6520 6d6f 6d65 2072 6174  And.the.mome.rat
	0x03d0:  6873 206f 7574 6772 6162 652e 0a         hs.outgrabe..
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 75)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0x85e4 (correct), seq 1:36, ack 936, win 8192, length 35
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  004b 0002 0000 4006 5c98 0a09 0807 0a01  .K....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 03a9 5010  ..............P.
	0x0030:  2000 85e4 0000 5468 6520 7175 6963 6b20  ......The.quick.
	0x0040:  6272 6f77 6e20 6672 6167 6d65 6e74 7320  brown.fragments.
	0x0050:  6a75 6d70 6564 206f 76                   jumped.ov
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 65)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0x7169 (correct), seq 36:61, ack 936, win 8192, length 25
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0041 0003 0000 4006 5ca1 0a09 0807 0a01  .A....@.\.......
	0x0020:  0203 0008 bdec 0000 0025 0000 03a9 5010  .........%....P.
	0x0030:  2000 7169 0000 6572 2074 6865 206c 617a  ..qi..er.the.laz
	0x0040:  7920 7061 636b 6574 732e 2054 6865 20    y.packets..The.
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 55)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0x6260 (correct), seq 61:76, ack 936, win 8192, length 15
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0037 0004 0000 4006 5caa 0a09 0807 0a01  .7....@.\.......
	0x0020:  0203 0008 bdec 0000 003e 0000 03a9 5010  .........>....P.
	0x0030:  2000 6260 0000 7175 6963 6b20 6272 6f77  ..b`..quick.brow
	0x0040:  6e20 6672 61                             n.fra
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [F.], cksum 0xafd5 (correct), seq 936, ack 76, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0004 0000 4006 5cb9 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 03a9 0000 004d 5011  .............MP.
	0x0030:  2000 afd5 0000                           ......
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [F.], cksum 0xafd4 (correct), seq 76, ack 937, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0005 0000 4006 5cb8 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 004d 0000 03aa 5011  .........M....P.
	0x0030:  2000 afd4 0000                           ......
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xafd4 (correct), ack 77, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0005 0000 4006 5cb8 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 03aa 0000 004e 5010  .............NP.
	0x0030:  2000 afd4 0000                           ......
